docs/UpgradeGuide.md

```diff
@@ -195,46 +195,96 @@ Datadog.configuration.ci[:cucumber][:service_name]
+Spans now inherit the global `service` name by default, unless otherwise explicitly set. This means, generally speaking, spans generated by Datadog integrations will default to the global `service` name, unless the `service_name` setting is configured for that integration.
 
-As an example, expect the following code & trace in 0.x:
+Spans that describe external services (e.g. `mysql`) will continue to default to some other name that describes the external service instead. (e.g. `mysql`)
 
 ```ruby
 ### Old 0.x ###
+Datadog.configure do |c|
+  # Instrumentation that measures internal behavior
+  c.use :rails, service_name: 'billing-api'
+  c.use :resque, service_name: 'billing-api'
+  c.use :sidekiq, service_name: 'billing-api'
+
+  # Instrumentation that measures external services
+  c.use :active_record, service_name: 'billing-api_mysql' # Defaults to DB type e.g. mysql
+  c.use :http, service_name: 'billing-api_http' # Defaults to net/http
+  c.use :redis, service_name: 'billing-api_redis' # Defaults to redis
+end
+
+### New 1.0 ###
 Datadog.configure do |c|
   c.service = 'billing-api'
-  c.use :rails
-  c.use :redis
-  c.use :resque
+
+  # Instrumentation that measures internal behavior
+  # now inherits the application's service name.
+  c.tracing.instrument :rails
+  c.tracing.instrument :resque
+  c.tracing.instrument :sidekiq
+
+  # Instrumentation that measures external services
+  # defaults to adapter-specific names. You may still override
+  # these names with the `service_name:` option.
+  c.tracing.instrument :active_record, service_name: 'billing-api_mysql' # Defaults to DB type e.g. mysql
+  c.tracing.instrument :http, service_name: 'billing-api_http' # Defaults to net/http
+  c.tracing.instrument :redis, service_name: 'billing-api_redis' # Defaults to redis
 end
 ```
 
-![0.x trace](./0.x-trace.png)
+**Update Rails instrumentation**
+
+If your application activates and configures `rails` instrumentation, you will need to adjust your settings slightly.
+
+The following options have been removed; instead, configure the underlying instrumentation directly.
 
-To reflect the following trace in 1.0 instead:
+| 0.x setting                                 | 1.0 setting                                                    |
+|---------------------------------------------|----------------------------------------------------------------|
+| `use :rails, cache_service: <SERVICE>`      | `tracing.instrument :active_support, cache_service: <SERVICE>` |
+| `use :rails, controller_service: <SERVICE>` | `tracing.instrument :action_pack, service_name: <SERVICE>`     |
+| `use :rails, database_service: <SERVICE>`   | `tracing.instrument :active_record, service_name: <SERVICE>`   |
+| `use :rails, job_service: <SERVICE>`        | `tracing.instrument :active_job, service_name: <SERVICE>`      |
+| `use :rails, log_injection: true`           | `tracing.log_injection = true`                                 |
```

Comment:
I think `log_injection` is try by default now

Reply:
You are right; this is superfluous.